src/execution_plan/optimizations/traverse_order.c

```diff
@@ -231,15 +241,74 @@ static void resolve_winning_sequence(AlgebraicExpression **exps, uint exp_count)
 	}
 }
 
+/* Having chosen which algebraic expression will be evaluated first, determine whether
+ * it is worthwhile to transpose it and thus swap the source and destination.
+ * We'll choose to transpose if the destination is filtered and the source is not, or
+ * if neither is filtered, if the destination is labeled and the source is not. */
```

Comment:
Updated the comment to clarify logic on bounded variables. Bound variables trump all other considerations, which I think for the moment is preferable, since it avoids introducing new non-redundant scan ops.

I think that the next iteration of the design here should quantify the value of different filters rather than operating on the boolean presence of a filter. An equality filter is of great value, an inequality filter is of minimal value.

If both `src` and `dest` are labeled and filtered, but `dest` can build an index scan, that's probably worth a transpose too!

I don't want to break with our current design more than necessary in this PR, though - let's try to write a new proposal soon.